test_case_gen.py

```diff
@@ -0,0 +1,249 @@
+"""
+Usage:
+    python2 test_case_gen.py
+
+Generates Topologies for Test Cases 0, 1, 2 each using three different
+congestion control algorithms (Dummy (always window size 1), Reno, FAST) and
+saves them to json.
+
+json's are stored in data/test_case_<number>_<flow type>.json
+"""
+from network_simulator.network_topology import *
+from network_simulator.host import *
+from network_simulator.common import *
+from network_simulator.router import *
+from network_simulator.flow import *
+from network_simulator.link import *
+
+if __name__ == "__main__":
+    # test case 0
+    links = list()
+    l1 = Link(end_1_addr = "H1", end_2_addr = "H2", link_buffer =

```

Comment:
Can you add a "name" field to Link's constructor and a corresponding ivar? For example, the name of this link would be "L1". That would be useful for Sharon later on.